spec/logical_volume_management_spec.rb

```diff
@@ -29,7 +29,8 @@
 
   describe "#setup" do
     before do
-      expect(@disk_double).to receive(:create_partition_table)
+      expect(@disk_double).to receive(:create_partition_table).with("gpt")
+      expect(@disk_double).to receive(:create_partition_table).with("msdos")
```

Comment:
@padajuan Noticed you asked about fixing the tests a week or so ago, and I have finally taken some time to give you an answer.

Without going into too much detail, what you need to do here is basically first make sure that you are stubbing the `#size` method on the `@disk_double`, since that wasn't expected to be be hit prior to your patch in the tests.  That said, you also want to make sure you are returning a size that would match what the existing tests were expecting previously, so something below the `2199023255552` you have in your patch.  The stub code should looks something like this:

```ruby
expect(@disk_double).to receive(:size).with(2000000000000)
```

The other thing is now that you have an variable argument for the `#create_partition_table` method, which you have started to stub, but unfortunately used a bit incorrectly.  What you have here will expect that method to be called once with the `"gpt"` arg and once with the `"msdos"` in **EACH** of the spec blocks below (the `it` blocks), instead of what you were probably intending, which was to allow it to accept either of the arguments.

I am attaching a diff on how you could go about this, as well as adding another test for the 2TB case.  I didn't get fancy on making this DRY-er, and there is a decent amount of copy-pasta involved, but it gets the job done.


```diff
diff --git a/spec/logical_volume_management_spec.rb b/spec/logical_volume_management_spec.rb
index 621ecda..077a97d 100644
--- a/spec/logical_volume_management_spec.rb
+++ b/spec/logical_volume_management_spec.rb
@@ -29,8 +29,6 @@ describe ManageIQ::ApplianceConsole::LogicalVolumeManagement do
 
   describe "#setup" do
     before do
-      expect(@disk_double).to receive(:create_partition_table).with("gpt")
-      expect(@disk_double).to receive(:create_partition_table).with("msdos")
       expect(@disk_double).to receive(:partitions).and_return([:fake_partition])
       @config.disk = @disk_double
 
@@ -55,6 +53,8 @@ describe ManageIQ::ApplianceConsole::LogicalVolumeManagement do
     end
 
     it "sets up the logical disk when mount point is not a symbolic link" do
+      expect(@disk_double).to receive(:size).and_return(2000000000000)
+      expect(@disk_double).to receive(:create_partition_table).with("msdos")
       @tmp_mount_point = @config.mount_point = Pathname.new(Dir.mktmpdir)
       expect(@fstab).to receive(:write!)
       expect(FileUtils).to_not receive(:mkdir_p).with(@config.mount_point)
@@ -71,6 +71,8 @@ describe ManageIQ::ApplianceConsole::LogicalVolumeManagement do
     end
 
     it "recreates the new mount point and sets up the logical disk when mount point is a symbolic link" do
+      expect(@disk_double).to receive(:size).and_return(2000000000000)
+      expect(@disk_double).to receive(:create_partition_table).with("msdos")
       @tmp_mount_point = Pathname.new(Dir.mktmpdir)
       @config.mount_point = Pathname.new("#{Dir.tmpdir}/#{@spec_name}")
       FileUtils.ln_s(@tmp_mount_point, @config.mount_point)
@@ -91,6 +93,8 @@ describe ManageIQ::ApplianceConsole::LogicalVolumeManagement do
     end
 
     it "skips update if mount point is in fstab when mount point is in already fstab" do
+      expect(@disk_double).to receive(:size).and_return(2000000000000)
+      expect(@disk_double).to receive(:create_partition_table).with("msdos")
       @tmp_mount_point = @config.mount_point = Pathname.new(Dir.mktmpdir)
       expect(FileUtils).to_not receive(:mkdir_p).with(@config.mount_point)
       expect(AwesomeSpawn).to receive(:run!)
@@ -108,5 +112,23 @@ describe ManageIQ::ApplianceConsole::LogicalVolumeManagement do
       expect(@config.logical_volume).to eq(@fake_logical_volume)
       expect(@fstab.entries.count).to eq(1)
     end
+
+    it "uses gpt partition table when disk size is over 2TB" do
+      expect(@disk_double).to receive(:size).and_return(2199023255552)
+      expect(@disk_double).to receive(:create_partition_table).with("gpt")
+      @tmp_mount_point = @config.mount_point = Pathname.new(Dir.mktmpdir)
+      expect(@fstab).to receive(:write!)
+      expect(FileUtils).to_not receive(:mkdir_p).with(@config.mount_point)
+      expect(AwesomeSpawn).to receive(:run!)
+        .with("mount",
+              :params => {"-t" => @config.filesystem_type, nil => ["/dev/vg_test/lv_test", @config.mount_point]})
+
+      @config.setup
+      expect(@config.partition).to eq(:fake_partition)
+      expect(@config.physical_volume).to eq(:fake_physical_volume)
+      expect(@config.volume_group).to eq(:fake_volume_group)
+      expect(@config.logical_volume).to eq(@fake_logical_volume)
+      expect(@fstab.entries.count).to eq(1)
+    end
   end
 end
```

Hope this helps.